pyfr/backends/cuda/compiler.py

```diff
@@ -103,18 +114,18 @@ def __init__(self, backend, src):
 
         # Compiler flags
         flags = [
-            f'--gpu-architecture=compute_{cmajor}{cminor}',
+            f'--gpu-architecture=sm_{cmajor}{cminor}',
             '--ftz=true',
             '--fmad=true'
         ]
 
         flags += shlex.split(backend.cfg.get('backend-cuda', 'cflags', ''))
 
-        # Compile to PTX
-        ptx = backend.nvrtc.compile('kernel', src, flags)
+        # Compile to CUDA code (either PTX or CUBIN, depending on arch flag)
+        cucode = backend.nvrtc.compile('kernel', src, flags)
 
         # Load it as a module
-        self.mod = backend.cuda.load_module(ptx)
+        self.mod = backend.cuda.load_module(cucode)
```

Comment:
Check if the `load_module` method needs its first argument changing from `ptx`.

Reply:
The docs for [cuModuleLoadDataEx](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__MODULE.html#group__CUDA__MODULE_1g9e8047e9dbf725f0cd7cafd18bfd4d12) indicate that it should take cubin or PTX interchangeably, but I think you're referring to just the variable naming convention, here. I'll refactor the references to PTX in `load_module` in order to avoid confusion.